src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Connection.cs

```diff
@@ -316,7 +324,16 @@ void IConnectionControl.Pause()
         void IConnectionControl.Resume()
         {
             Log.ConnectionResume(ConnectionId);
-            _socket.ReadStart(_allocCallback, _readCallback, this);
+            try
+            {
+                _socket.ReadStart(_allocCallback, _readCallback, this);
```

Comment:
I think I disagree with you about checking `File.Exists`, at least when there's a good chance that the file in fact doesn't exist. This is all happening on the libuv thread so there is no race going on like there might be with files. The only time ReadStart may throw as far as I'm aware is in the somewhat rare case that we called ReadStop after receiving the last bit of data but before the FIN/reset.

It might be worth asking the libuv mailing list to see if there is a way to listen for a connection closed event without relying on the read callback (or somehow still applying back pressure).

If there is no other way, we could arguably have ReadStart return the status. We would likely have to check it and throw it it fails during the first call in Connection.Start, but here we could just check it in an if instead of using try/catch for control flow.